lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -109,14 +110,23 @@ def test_not_a_file(self, query_get_as, monkeypatch):
 
     def test_dataset_in_given_path(self, query_get_as, monkeypatch):
         mock_args: Optional[tuple[Path, dict[str, Any]]] = None
+        exp_stream = io.BytesIO(b"file_as_a_byte_stream")
+        filestream_dict = {
+            "name": "f1.json",
+            "type": CacheType.FILE,
+            "stream": exp_stream,
+        }
 
         def mock_send_file(path_or_file, *args, **kwargs):
             nonlocal mock_args
+            assert path_or_file == exp_stream
             mock_args = (path_or_file, kwargs)
             return {"status": "OK"}
 
         monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
-        monkeypatch.setattr(Path, "is_file", lambda self: True)
+        monkeypatch.setattr(
+            self.MockTarball, "filestream", lambda _s, _t: filestream_dict
+        )
```

Comment:
Now that we're mocking `filestream()`, do we still need the mock for `find_dataset()`?

Reply:
yes, else it is unable to find the dataset and fails at the very first step.

Reply:
OK, I see the (my) problem:  there's still some confusion about what "the unit" is here that we're testing.  I was expecting it to be `DatasetsInventory._get()`, but you're including `CacheManager.filestream()` in it.  That is, you're mocking `Tarball.filestream()` where I think we should be mocking `CacheManager.filestream()`. 

`DatasetsInventory._get()` doesn't call `find_dataset()` anymore -- you've replaced that call with a call to `CacheManager.filestream()` -- if we mock `CacheManager.filestream()`, then we don't need to mock either `CacheManager.find_dataset()` or `Tarball.filestream()`.

We _shouldn't_ need to include `CacheManager.filestream()` in this unit, because we should be testing it separately.  However, given that the function doesn't contain any conditionals or perform any calculations or transformations, I think it's OK to leave it untested -- is just a wrapper which contains straight-line code (although, that also means that it's very easy to write a test for, and it only has one scenario...).